### ISSUE TITLE
AttributeError thrown upon every autocompletion trigger, when file contains a certain import

### ISSUE THREAD

**Issue by bradst** [zchee/deoplete-jedi#30]
# Problems summary
When editing a file that contains `from decimal import Decimal` at the top, AttributeError is thrown every time autocompletion is triggered.
```
[deoplete] Traceback (most recent call last):
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 47, in completion_begin
[deoplete]     complete_position, candidates = self.gather_candidates(context)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 85, in gather_candidates
[deoplete]     results = self.gather_results(context)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 150, in gather_results
[deoplete]     context['candidates'] = source.gather_candidates(context)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/sources/deoplete_jedi.py", line 342, in gather_candidates
[deoplete]     name, type_, desc, abbr = self.parse_completion(c, tmp_filecache)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/sources/deoplete_jedi.py", line 184, in parse_completion
[deoplete]     type_, desc = [x.strip() for x in comp.description.split(':', 1)]
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/api/classes.py", line 425, in description
[deoplete]     t = self.type
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/api/classes.py", line 470, in type
[deoplete]     followed = self.follow_definition()
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/cache.py", line 41, in wrapper
[deoplete]     rv = function(obj, *args, **kwargs)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/api/classes.py", line 498, in follow_definition
[deoplete]     defs = self._follow_statements_imports()
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/cache.py", line 41, in wrapper
[deoplete]     rv = function(obj, *args, **kwargs)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/api/classes.py", line 485, in _follow_statements_imports
[deoplete]     return i.follow()
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/cache.py", line 41, in wrapper
[deoplete]     rv = function(obj, *args, **kwargs)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/imports.py", line 103, in follow
[deoplete]     for t in types))
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/imports.py", line 103, in <genexpr>
[deoplete]     for t in types))
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/__init__.py", line 120, in find_types
[deoplete]     return f.find(scopes, search_global)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/debug.py", line 52, in wrapper
[deoplete]     result = func(*args, **kwargs)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/finder.py", line 86, in find
[deoplete]     names = self.filter_name(scopes)
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/finder.py", line 176, in filter_name
[deoplete]     for names_dict, position in names_dicts:
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/finder.py", line 109, in <genexpr>
[deoplete]     return ((n, None) for n in self.scope.names_dicts(search_global))
[deoplete]   File "/home/brad/.config/nvim/plugged/deoplete-jedi/rplugin/python3/deoplete/jedi/jedi/evaluate/representation.py", line 720, in names_dicts
[deoplete]     yield star_module.names_dict
[deoplete] AttributeError: 'NoneType' object has no attribute 'names_dict'
[deoplete] An error has occurred. Please execute :messages command.
```

## Expected
Auto-completion should function correctly and not throw exceptions.

## Environment Information

 * OS: Arch Linux
 * Neovim version: 0.1.2

## Provide a minimal init.vim with less than 50 lines (Required!)

```vim
call plug#begin('~/.config/nvim/plugged')
Plug 'Shougo/deoplete.nvim'
Plug 'zchee/deoplete-jedi'
call plug#end()
let g:deoplete#enable_at_startup=1
```

## The reproduce ways from neovim starting (Required!)

 1. Open a new or existing Python file.
 2. At the top of the file, add `from decimal import Decimal` .
 3. Start typing on a new line until auto-completion triggers (2 characters by default). Auto-completion will not function and AttributeError will be shown under `:messages` .

## Generate a logfile if appropriate

[jedi_debug.txt](https://github.com/zchee/deoplete-jedi/files/211016/jedi_debug.txt)

## Screen shot (if possible)

![2016-04-08_20-15-43](https://cloud.githubusercontent.com/assets/18357704/14400523/3c3ed8a0-fdc7-11e5-8eaa-9a92b1fb943a.png)

**Comment by zchee** [comment]
@bradst Hi,

I do not yet know the cause of the error.
but, Now neovim stable version is `0.1.3`. Could you upgrade it?

Also, What are you using `neovim/python-client`?

**Comment by bradst** [comment]
@zchee No problem. I've just upgraded neovim and python-client to latest git. neovim is now 0.1.3 (git HEAD 5330aa1) and python-client is now 0.1.7 (git HEAD 7af000e). The exception is still being thrown with the new version, exact same traceback.

**Comment by tweekmonster** [comment]
@bradst We've been finding that a lot of the issues with deoplete-jedi is stemming from Jedi and different Python versions.  Take a look at this issue to get up to speed on what I'm talking about: https://github.com/zchee/deoplete-jedi/issues/21

We're currently hashing out a fix for most of the Python version issues.  You can try out this branch to see if your issue goes away: https://github.com/tweekmonster/deoplete-jedi/tree/server

I'm currently on that branch and am able to do the Decimal completion without errors.  You can also help us hash out issues with that branch 😄

**Comment by bradst** [comment]
@tweekmonster I just tried your branch. No exceptions are thrown, but `from decimal import Decimal` still causes auto-completion not to work.

I just did some digging and figured it out. The `decimal` module tries to use the native `_decimal` module and falls back to `_pydecimal` on ImportError. Apparently the native module depends on an external library called `mpdecimal`, which I didn't have installed, so Python was falling back on the pure Python implementation. It seems like Jedi wasn't able to follow the fallback so it choked when trying to access the native module, or something similar. Installing `mpdecimal` fixed the issue.

Thanks guys, I'll report this upstream to Jedi. Keep up the good work! :smiley:
